main.py

```diff
@@ -54,17 +55,22 @@ def main(config):
 
     # Sample from trained GFlowNet
     if config.n_samples > 0 and config.n_samples <= 1e5:
-        samples, times = gflownet.sample_batch(env, config.n_samples, train=False)
-        energies = env.oracle(env.statebatch2oracle(samples))
+        states, times = gflownet.sample_batch(env, config.n_samples, train=False)
+        samples = env.statebatch2oracle(states)
+        energies = env.oracle(samples)
+        gflownet.evaluate(samples, energies)
         df = pd.DataFrame(
             {
-                "readable": [env.state2readable(s) for s in samples],
-                "energies": energies.tolist(),
+                "readable": [env.state2readable(s) for s in states],
+                "energies": energies.cpu().numpy(),
```

Comment:
The energies will always be a one dimensional "array", that is a list. So maybe it's better to change the output of the oracle to make it work with `tensor.tolist()`? What do you think?

Reply:
Ah yes makes sense. The output of the oracle is a tensor itself. I don't remember what made me change it to numpy format before saving here, I will revert this to `tolist()`